### PR TITLE
batches: migrate execution page requests to Apollo

### PR DESCRIPTION
**Pull request by courier-new**
Closes https://github.com/sourcegraph/sourcegraph/issues/32343.

Migrates queries and mutations called on the execution screen, with the exception of file diff connection queries, to Apollo Client and `useQuery` hook.

Don't be alarmed by the diff stat -- ~300 lines of that is in mock data and the storybook story. This PR is net negative lines of code without them. 🙂

## Test plan

Introduced additional storybook coverage for `WorkspaceDetails` in different states. Tested requests and polling manually.

## App preview:

- [Web](https://sg-web-kr-execution-requests-apollo.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-xgoxnwryxu.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/batches/execution/BatchSpecExecutionDetailsPage.tsx [hunk 1]
```diff
@@ -276,17 +262,12 @@ const BatchSpecActions: React.FunctionComponent<BatchSpecActionsProps> = ({ batc
...
                             outline={true}
                             variant="danger"
                         >
-                            {isCanceling !== true && <>Cancel</>}
-                            {isCanceling === true && (
-                                <>
-                                    <LoadingSpinner /> Canceling
-                                </>
-                            )}
+                            {isCancelLoading ? <LoadingSpinner /> : 'Cancel'}
```
Comment (eseliger) [inline client/web/src/enterprise/batches/execution/BatchSpecExecutionDetailsPage.tsx:270]: did you intentionally remove the button label here?

File: client/web/src/enterprise/batches/execution/BatchSpecExecutionDetailsPage.tsx [hunk 2]
```diff
@@ -300,18 +281,13 @@ const BatchSpecActions: React.FunctionComponent<BatchSpecActionsProps> = ({ batc
...
                             outline={true}
                             variant="secondary"
                         >
-                            {isRetrying !== true && <>Retry</>}
-                            {isRetrying === true && (
-                                <>
-                                    <LoadingSpinner /> Retrying
-                                </>
-                            )}
+                            {isRetryLoading ? <LoadingSpinner /> : 'Retry'}
```
Comment (eseliger) [inline client/web/src/enterprise/batches/execution/BatchSpecExecutionDetailsPage.tsx:290]: Same question as above

**Review by eseliger — APPROVED** [review]
Thanks for all the additional stories :) Sad they can't be covered by chromatic 🙃

File: client/web/src/enterprise/batches/execution/BatchSpecExecutionDetailsPage.tsx [hunk 3]
```diff
@@ -300,18 +281,13 @@ const BatchSpecActions: React.FunctionComponent<BatchSpecActionsProps> = ({ batc
...
                             outline={true}
                             variant="secondary"
                         >
-                            {isRetrying !== true && <>Retry</>}
-                            {isRetrying === true && (
-                                <>
-                                    <LoadingSpinner /> Retrying
-                                </>
-                            )}
+                            {isRetryLoading && <LoadingSpinner />}Retry
```
Comment (eseliger) [inline client/web/src/enterprise/batches/execution/BatchSpecExecutionDetailsPage.tsx:290]: Isn't this missing a space in between the loader and the retry now?

Comment (courier-new) [reply client/web/src/enterprise/batches/execution/BatchSpecExecutionDetailsPage.tsx:290]: Good question, we seem to inconsistently put a space here...

**Has a space**
- `BatchChangeCloseAlert`

**Doesn't have a space**
- `AddCredentialModal`
- `RemoveCredentialModal`
- the various bulk operation modals
- `BatchChangeCloseAlert`

So I went with the majority and omitted it. 😂

Searching a bit more broadly, we actually have a component called a `LoaderButton` now that sticks the spinner in consistently on the left side and applies a left margin to text instead, so perhaps that's what these should all be. 🤔 Maybe I'll refactor over our company meeting. 🙂

Comment (courier-new) [reply client/web/src/enterprise/batches/execution/BatchSpecExecutionDetailsPage.tsx:290]: https://github.com/sourcegraph/sourcegraph/pull/34519

**Pull request merged**